### PR TITLE
[enterprise-4.12] TELCODOCS-1838-412 Update reference SiteConfig to use rootDeviceHints 

### PR DESCRIPTION
**Pull request by kquinn1204**
[TELCODOCS-1838](https://issues.redhat.com//browse/TELCODOCS-1838)]: Docs: Create partitions through SiteConfig

Version(s):
4.12

Issue:https://issues.redhat.com/browse/TELCODOCS-1832

Link to docs preview:

- https://80285--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.html
- https://80285--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-sites.html
- https://80285--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-precaching-tool.html

QE review:
- [x] QE has approved this change.

Additional information:
“Cherry Picked from d59e7cf02bd625fb56cd46e16d91b48c6859a07b xref: [https://github.com/openshift/openshift-docs/pull/75726]

### CODE REVIEW

**Review by adellape — COMMENTED** [review]
Looks like this has pretty much all the same notes I would make from the 4.13 version of this cherrypick, so refer to my comments there for this PR as well: https://github.com/openshift/openshift-docs/pull/80284#pullrequestreview-2278414565

**Comment by kquinn1204** [comment]
cheers @adellape

File: modules/ztp-configuring-disk-partitioning.adoc [hunk 1]
```diff
@@ -6,41 +6,197 @@
 [id="ztp-configuring-disk-partitioning_{context}"]
 = Configuring disk partitioning with SiteConfig
 
-Configure disk partitioning for a managed cluster using a `SiteConfig` CR and GitOps ZTP. The  disk partition details in the `SiteConfig` CR must match the underlying disk.
+Configure disk partitioning for a managed cluster using a `SiteConfig` CR and {ztp-first}. The  disk partition details in the `SiteConfig` CR must match the underlying disk.
```
Comment (bscott-rh) [inline modules/ztp-configuring-disk-partitioning.adoc:9]: This attribute is not rendering in the preview. It looks like the `{ztp-first}` attribute is missing from the `_attributes/common-attributes.adoc` file in `enterprise-4.12`. Can you add it in this PR?

Comment (kquinn1204) [reply modules/ztp-configuring-disk-partitioning.adoc:9]: good catch added

**Comment by kquinn1204** [comment]
Thanks @bscott-rh hopefully OK Now for merge

**Comment by bscott-rh** [comment]
Merge review LGTM.

**Pull request merged**